pkg/netpol/eval/resources.go

```diff
@@ -582,29 +583,31 @@ func (pe *PolicyEngine) ConvertPeerNamedPort(namedPort string, peer Peer) (int32
 // the pe.podsMap in case of fake ingress-controller pods
 // or the pe.representativePeersMap in case of exposure-analysis peers
 func (pe *PolicyEngine) AddPodByNameAndNamespace(name, ns string, objLabels *k8s.SingleRuleLabels) (Peer, error) {
-	var nsLabels map[string]string
+	var nsLabels, podLabels map[string]string
 	if objLabels != nil {
 		nsLabels = objLabels.NsLabels
+		podLabels = objLabels.PodLabels
 	}
 
 	podStr := types.NamespacedName{Namespace: ns, Name: name}.String()
 	newPod := &k8s.Pod{
 		Name:      name,
 		Namespace: ns,
 		FakePod:   true,
+		Labels:    podLabels,
 	}
 	if err := pe.resolveSingleMissingNamespace(ns, nsLabels); err != nil {
 		return nil, err
 	}
-	if pe.exposureAnalysisFlag && newPod.Name == k8s.RepresentativePodName { // if exposure-analysis and this is not a fake ingress-controller
+	// if exposure-analysis and this is not a fake ingress-controller
+	if pe.exposureAnalysisFlag && strings.HasPrefix(newPod.Name, k8s.RepresentativePodName) {
 		// first compute a unique string from labels to be used as a map key
-		keyStrFromNsLabels := k8s.VariantFromLabelsMap(objLabels.NsLabels)
-		// todo : when supporting also pod labels; add also key str from pod labels to the map key
-		if _, ok := pe.representativePeersMap[keyStrFromNsLabels]; ok { // we already have a representative peer with same labels
+		keyStrFromLabels := k8s.VariantFromLabelsMap(objLabels.NsLabels) + k8s.VariantFromLabelsMap(objLabels.PodLabels)
```

Comment:
add a delimiter for concatenation, to be on the safe side that no two objects are mapped to the same key

Reply:
done